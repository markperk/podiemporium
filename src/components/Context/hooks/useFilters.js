import { useBrix } from 'react-brix'
import { fromJS } from 'immutable'
import { paths } from '../paths'

export const useFilters = () => {
  const [genre, setGenre] = useBrix(paths.filters.genre.get())
  const [playId, setPlayId] = useBrix(paths.filters.playId.get())
  const [rating, setRating] = useBrix(paths.filters.rating.get())
  const [year, setYear] = useBrix(paths.filters.year.get())
  const range = rating && [rating.get(0), rating.get(1)]

  let filters = {}
  if (genre) filters.genre = [genre]
  if (playId) filters.play_id = [playId]
  if (year) filters.year = [year]
  if (rating) filters.rating = range

  return [
    fromJS(filters),
    {
      useGenre: [genre, setGenre],
      usePlayId: [playId, setPlayId],
      useRating: [range, setRating],
      useYear: [year, setYear],
    }
  ]
}