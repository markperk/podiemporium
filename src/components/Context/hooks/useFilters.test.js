import { useBrix } from 'react-brix'
import { useFilters } from './useFilters'
import { List } from 'immutable'

jest.mock('react')
jest.mock('react-brix')

describe('useFilters', () => {

  describe('genre', () => {
    let useGenreMock
    beforeEach(() => {
      jest.clearAllMocks()

      // eslint-disable-next-line
      Array.prototype.get = jest.fn(x => 0)
      useGenreMock = [['tragedy'], jest.fn()]
      useBrix.mockReturnValue(useGenreMock)
    })

    test('returns filters, hooks', () => {
      const [filters, { useGenre }] = useFilters()
      expect(filters).toMatchSnapshot()
      expect(useGenre).toEqual(useGenreMock)
    })
  })

  describe('play_id', () => {
    let usePlayIdMock
    beforeEach(() => {
      jest.clearAllMocks()

      // eslint-disable-next-line 
      String.prototype.get = jest.fn(x => 0)
      usePlayIdMock = ['1', jest.fn()]
      useBrix.mockReturnValue(usePlayIdMock)
    })

    test('returns filters, hooks', () => {
      const [filters, { usePlayId }] = useFilters()
      expect(filters).toMatchSnapshot()
      expect(usePlayId).toEqual(usePlayIdMock)
    })
  })

  describe('rating', () => {
    let useRatingMock
    beforeEach(() => {
      jest.clearAllMocks()

      useRatingMock = [List([1, 2]), jest.fn()]
      useBrix.mockReturnValue(useRatingMock)
    })

    test('returns filters, hooks', () => {
      const [filters, { useRating }] = useFilters()
      const expected = JSON.stringify([[1, 2], jest.fn()])
      const result = JSON.stringify(useRating)
      expect(filters).toMatchSnapshot()
      expect(result).toEqual(expected)
    })
  })
})
