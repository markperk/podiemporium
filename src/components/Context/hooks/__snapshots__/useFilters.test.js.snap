// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFilters genre returns filters, hooks 1`] = `
Immutable.Map {
  "genre": Immutable.List [
    Immutable.List [
      "tragedy",
    ],
  ],
  "play_id": Immutable.List [
    Immutable.List [
      "tragedy",
    ],
  ],
  "rating": Immutable.List [
    0,
    0,
  ],
}
`;

exports[`useFilters play_id returns filters, hooks 1`] = `
Immutable.Map {
  "genre": Immutable.List [
    "1",
  ],
  "play_id": Immutable.List [
    "1",
  ],
  "rating": Immutable.List [
    0,
    0,
  ],
}
`;

exports[`useFilters rating returns filters, hooks 1`] = `
Immutable.Map {
  "genre": Immutable.List [
    Immutable.List [
      1,
      2,
    ],
  ],
  "play_id": Immutable.List [
    Immutable.List [
      1,
      2,
    ],
  ],
  "rating": Immutable.List [
    1,
    2,
  ],
}
`;
